src/lib/sqlite/sqliterg.ts

```diff
@@ -0,0 +1,351 @@
+          'Content-Type': 'application/json',
+          'Authorization': `Basic ${authBase64}`,
+        },
+        body: JSON.stringify(request)
+      },
+    )
+    return response.json()
+  }
+  return {
+    async isReachable(): Promise<boolean> {
+      try {
+        const response = await fetch(
+          url,
+          {
+            method: 'HEAD',
+            mode: 'no-cors',
+          }
+        )
+        return true
+      } catch (e) {
+        return false
+      }
+    },
+    async isAuthenticated(): Promise<boolean> {
+      try {
+        const response = await send({
+          transaction: [
+            {
+              query: 'SELECT 1'
+            }
+          ],
+        })
+        if ('results' in response) {
+          return response.results[0].success
+        }
+        if ('reqIdx' in response) {
+          console.warn('createSqlitergExecutor.isAuthenticated', response)
+          return false
+        }
+      } catch (e) {
+        return false
+      }
+      throw Error('isAuthenticated: unreachable code')
+    },
+    async execute(queryOrStatement: string, params: Params | any[]): Promise<Response> {
+      const queryOrStatementLowered = queryOrStatement.toLowerCase()
+      let request: Request = {
+        transaction: [],
+      }
+
+      if (
+        queryOrStatementLowered.startsWith('select') ||
+        queryOrStatementLowered.startsWith('pragma')
+      ) {
+        const transaction: Transaction = {
+          query: queryOrStatement,
+          values: params,
+        }
+        request.transaction.push(transaction)
+      } else {
+        const transaction: Transaction = {
+          statement: queryOrStatement,
+          values: params,
+        }
+        request.transaction.push(transaction)
+      }
+
+      return await send(request)
+    }
+  }
+}
+
+export function createRemoteDb(executor: SqlitergExecutor) {
+  return drizzle(async (queryString, params, method) => {
+    const response = await executor.execute(queryString, params)
+    if ('reqIdx' in response) {
+      console.warn('createRemoteDb.drizzle execution ', response)
+      return {rows: []}
+    }
+    if (response.results.length === 0 || !response.results[0].success) {
+      return {rows: []}
+    }
+    const result = response.results[0]
+    // @ts-ignore
+    const records = result['resultSet'] as any[]
+    if (records) {
+      const values = records.map(Object.values)
+      if (method === 'get' && values.length > 0) {
+        return {rows: values[0]}
+      }
+      return {rows: values}
+    }
+    return {rows: []}
+  })
+}
+
+export type RemoteSnippetStore =
+  SnippetStore &
+  {
+    clear(): Promise<void>
+    isAvailable(): Promise<boolean>
+    refresh(): Promise<void>
+  }
+
+export async function createRemoteSnippetStore(
+  migrationStateStore: Writable<MigrationState>,
+  globalStateStore: Writable<GlobalState>,
+  executorStore: Readable<SqlitergExecutor>,
```

Comment:
Must we split `migrationStateStore` out like this? `migrate` is idempotent anyway. We can just move running `migrate` to within `createRemoteSnippetStore` or `createLocalSnippetStore` as the store would exist once throughout the application's life anyway.